pkg/network/encoding/marshal/format.go

```diff
@@ -287,9 +286,10 @@ func formatTags(c network.ConnectionStats, tagsSet *network.TagsSet, connDynamic
 	// other tags, e.g., from process env vars like DD_ENV, etc.
 	for tag := range c.Tags {
 		mm.Reset()
-		_, _ = mm.Write(unsafeStringSlice(tag))
+		t := tag.Get().(string)
+		_, _ = mm.Write(unsafeStringSlice(t))
```

Comment:
If your benchmarks will let you, it'd be really interesting to see if go escape analysis will do the right thing if you use `[]byte(t)`

Reply:
`[]byte(t)` seems to perform worse:
```
goos: linux
goarch: amd64
pkg: github.com/DataDog/datadog-agent/pkg/network/encoding/marshal
cpu: 11th Gen Intel(R) Core(TM) i9-11950H @ 2.60GHz
             │   old.txt   │               new.txt               │
             │   sec/op    │   sec/op     vs base                │
FormatTags-4   237.9n ± 1%   288.8n ± 1%  +21.42% (p=0.000 n=10)

             │  old.txt   │              new.txt               │
             │    B/op    │    B/op     vs base                │
FormatTags-4   104.0 ± 0%   144.0 ± 0%  +38.46% (p=0.000 n=10)

             │  old.txt   │               new.txt               │
             │ allocs/op  │ allocs/op   vs base                 │
FormatTags-4   3.000 ± 0%   6.000 ± 0%  +100.00% (p=0.000 n=10)
```